analyses/cell-type-consensus/scripts/04-assign-consensus-celltypes.R

```diff
@@ -0,0 +1,171 @@
+# cellassign_celltype_annotation: Original cell type name from CellAssign (panglao name)
+# panglao_ontology: CL term assigned to panglao term
+# panglao_annotation: human readable value associated with the CL term for panglao term
+# blueprint_annotation_cl: human readable value associated with the CL term for singler_celltype_ontology
+# consensus_annotation: human readable name associated with the consensus label
+# consensus_ontology: CL ontology term for the consensus cell type
+
+library(optparse)
+
+option_list <- list(
+  make_option(
+    opt_str = c("--sce_file"),
+    type = "character",
+    help = "Path to RDS file containing a processed SingleCellExperiment object from scpca-nf"
+  ),
+  make_option(
+    opt_str = c("--blueprint_ref_file"),
+    type = "character",
+    help = "Path to file with BlueprintEncodeData cell ontology IDs and associated cell ontology names"
+  ), 
+  make_option(
+    opt_str = c("--panglao_ref_file"),
+    type = "character",
+    help = "Path to file with panglao assignments and associated cell ontology ids"
+  ),
+  make_option(
+    opt_str = c("--consensus_ref_file"),
+    type = "character",
+    help = "Path to file containing the reference for assigning consensus cell type labels"
+  ),
+  make_option(
+    opt_str = c("--output_file"),
+    type = "character",
+    help = "Path to file where combined TSV file will be saved.
+      File name must end in either `.tsv` or `.tsv.gz` to save a compressed TSV file"
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Set up -----------------------------------------------------------------------
+
+# make sure input files exist
+stopifnot(
+  "sce file does not exist" = file.exists(opt$sce_file),
+  "blueprint reference file does not exist" = file.exists(opt$blueprint_ref_file),
+  "panglao reference file does not exist" = file.exists(opt$panglao_ref_file),
+  "cell type consensus reference file does not exist" = file.exists(opt$consensus_ref_file),
+  "output file must end in `.tsv` or `.tsv.gz`" = stringr::str_detect(opt$output_file, ".tsv|.tsv.gz")
+)
+
+# load SCE
+suppressPackageStartupMessages({
+  library(SingleCellExperiment)
+})
+
+# Extract colData --------------------------------------------------------------
+
+# read in sce 
+sce <- readr::read_rds(opt$sce_file)
+
+# extract ids 
+library_id <- metadata(sce)$library_id
+# account for multiplexed libraries that have multiple samples 
+# for now just combine sample ids into a single string and don't worry about demultiplexing 
+sample_id <- metadata(sce)$sample_id |> 
+  paste0(collapse = ";")
+project_id <- metadata(sce)$project_id
+
+# check if cell line since cell lines don't have any cell type assignments
+# account for having more than one sample and a list of sample types
+# all sample types should be the same theoretically
+sample_type <- unique(metadata(sce)$sample_type)
+is_cell_line <- sample_type == "cell line"
```

Comment:
This can now be cell line and pdx, right? Do pdx cell lines have assignments?

Reply:
Ohh good catch. And no they don't. I'll update this to check for either just `cell line` or the combo.